lib/Runtime/Types/Type.h

```diff
@@ -62,6 +63,14 @@ namespace Js
         inline BOOL SkipsPrototype() const { return (this->flags & TypeFlagMask_SkipsPrototype) != 0 ; }
         inline BOOL CanHaveInterceptors() const { return (this->flags & TypeFlagMask_CanHaveInterceptors) != 0; }
         inline BOOL IsFalsy() const { return flags & TypeFlagMask_IsFalsy; }
+        inline BOOL HasBeenCached() const { return flags & TypeFlagMask_HasBeenCached; }
+        inline void SetHasBeenCached(const bool hasBeenCached)
+        {
+            // Once set, this flag should never be reset.
```

Comment:
Types are invalidated? Do you mean caches are invalidated? If we want to optimize those cases, the tracking becomes much harder - we'll basically have to somehow track all the caches in which a type has been cached, then reset this flag once all of those caches have been invalidated. (When an inline cache gets invalidated, there may still be others which hold on to the type)

---
In reply to: [78846360](https://github.com/Microsoft/ChakraCore/pull/1542#discussion_r78846360) [](ancestors = 78846360,78828446,78818005,78816821,78790053)